### PR TITLE
Replace request with got release-post script

### LINKED ISSUE #2689: Replace `request` with another package?
We use request in two places, one is download.js which will be removed in #2688 and the other one is in https://github.com/nodejs/nodejs.org/blob/master/scripts/release-post.js.

Note that this won't remove any deps, since request is a node-sass dependency.

### PR DESCRIPTION
**Pull request by MrBenJ**
Fixes #2689

### CODE REVIEW

File: scripts/release-post.js [hunk 1]
```diff
@@ -33,15 +33,14 @@ function sendRequest (opts) {
       headers: { 'User-Agent': 'nodejs.org release blog post script' }
     }, opts)
 
-    request(options, (err, res, body) => {
-      if (err) {
-        return reject(new Error(`Error requesting URL ${options.url}: ${err.message}`))
+    const { url, ...gotOptions } = options
```
Comment (XhmikosR) [inline scripts/release-post.js]: Is the destructuring really needed here?

**Comment by XhmikosR** [comment]
Need a rebase also :)

**Comment by XhmikosR** [comment]
BTW would be nice to use the same for download.js and thus remove request completely from the project dependencies. Although, I have a PR which removes the file in #2688

File: scripts/release-post.js (hunk 1 above)
Comment (MrBenJ) [reply scripts/release-post.js]: It isn't. I was just trying to be safe with the options if url was passed in twice. I took it out in another commit, along with a rebase to master :). Let me know what you think!

**Comment by MrBenJ** [comment]
@XhmikosR - I was thinking of that too, but knowing the file is about to be deleted, I think it would be better just to leave that in. Maybe in your previous PR you can delete `request` from `package.json` since it's still in here? PR race? ;)

If that PR gets merged in first, I'll be happy to rebase again and remove request completely from the dependencies.

Let me know what's up!

**Comment by MrBenJ** [comment]
@XhmikosR  - any news?

**Comment by XhmikosR** [comment]
It's not my decision to make only :)

/CC @Trott @MylesBorins for review. Assuming we move forward with #2688, this is a good step forward.

**Comment by Trott** [comment]
@nodejs/website

**Comment by XhmikosR** [comment]
@MrBenJ since #2688 might not even be merged or take a long time, can you update download.js to use got too? BTW I've rebased the branch, so be careful not to overwrite it.

**Comment by silverwind** [comment]
I guess it's mostly down to personal preference, but I prefer a `fetch` implementation over `got`:

https://www.npmjs.com/package/make-fetch-happen
https://www.npmjs.com/package/node-fetch

The former offers proxy support just like `request`, so it is my go-to fetching library currently.

**Comment by XhmikosR** [comment]
FYI I don't mind what we use, as long as we do it for all scripts and thus replace request :)

**Comment by XhmikosR** [comment]
@MrBenJ can you rebase this and also update to the latest got version? The other script that was using request is now removed, so you can remove request completely from the repo.

**Comment by XhmikosR** [comment]
Alternatively, if someone has time, we could just use the native https to do the same and not add a dependency.

**Comment by MrBenJ** [comment]
Hey @XhmikosR ! I'm on holiday right now but have the time off tomorrow to probably dig into this. I think I'll make another fork of the repo and just start fresh with all this new info :).

**Comment by XhmikosR** [comment]
Sound good to me!

**Pull request closed without merging**